consensus/types/round_state.go

```diff
@@ -80,6 +80,9 @@ type RoundState struct {
 	LockedBlock        *types.Block        `json:"locked_block"`
 	LockedBlockParts   *types.PartSet      `json:"locked_block_parts"`
 
+	// Set of block hashes already prevoted by this validator in this round
+	PrevotedHashes []string `json:"prevoted_hashes"`
```

Comment:
for efficiency this could be a `map[string]struct{}` (i.e. a set of hashes), however I wasn't sure if this needs to be jsonified